hy/models.py

```diff
@@ -121,22 +121,39 @@ def __init__(self, string):
 _wrappers[type(None)] = lambda foo: HySymbol("None")
 
 
-class HyKeyword(HyObject, str_type):
+class HyKeyword(HyObject):
     """Generic Hy Keyword object. It's either a ``str`` or a ``unicode``,
     depending on the Python version.
     """
 
-    PREFIX = "\uFDD0"
+    __slots__ = ['_value']
 
-    def __new__(cls, value):
-        if not value.startswith(cls.PREFIX):
-            value = cls.PREFIX + value
-
-        obj = str_type.__new__(cls, value)
-        return obj
+    def __init__(self, value):
+        if value[0] != ':':
+            value = ':' + value
+        self._value = value
 
     def __repr__(self):
-        return "%s(%s)" % (self.__class__.__name__, repr(self[1:]))
+        return "%s(%r)" % (self.__class__.__name__, self._value)
+
+    def __str__(self):
+        return self._value
+
+    def __hash__(self):
+        return hash(self._value)
+
+    def __eq__(self, other):
+        if not isinstance(other, HyKeyword):
+            return NotImplemented
```

Comment:
That's actually got different semantics, as I understand it.

Because, abstractly, when running `a == b`, python tried first `a.__eq__(b)`, and if `NotImplemented` is returned, it'll try `b.__eq__(a)`.  Returning `False` here means we'll break code that might want to test equality with a `HyKeyword` (I don't know why you would, but I don't think that justifies breaking that intentionally).

Reply:
Oh, you're right, I think. With your code, `(= :hello "hello")` returns false instead of throwing an error.